src/js.c

```diff
@@ -2559,8 +2598,15 @@ _subscribe(natsSubscription **new_sub, jsCtx *js, const char *subject, const cha
                     s = nats_setDefaultError(NATS_NO_MEMORY);
             }
             IF_OK_DUP_STRING(s, jsi->stream, stream);
-            if ((s == NATS_OK) && !nats_IsStringEmpty(subject))
-                DUP_STRING(s, jsi->psubj, subject);
+            if ((s == NATS_OK) && numSubjects > 0)
+            {
+                int l = nats_printStringArray(NULL, 0, subjects, numSubjects);
```

Comment:
First I think the `print` is a bit confusing, but why not go with a simple `natsBuf` and just call append with each string/byte for the comma, the same way we do when marshaling?